### ISSUE TITLE
Error while highlighting the buffer for `current_buffer_fuzzy_find`

### ISSUE THREAD

**Issue by dhruvmanila** [nvim-telescope/telescope.nvim#1151]
### Description

As mentioned in this comment (https://github.com/nvim-telescope/telescope.nvim/issues/795#issuecomment-903281639), a change in the Neovim core is creating this error.

```
Error executing vim.schedule lua callback: .../opt/telescope.nvim/lua/telescope/pickers/highlights.lua:32: Expected lua string
```

### Neovim version

NVIM v0.6.0-dev+159-g2460f0a70 Build type: Release LuaJIT 2.1.0-beta3 Compilation: /Library/Developer/CommandLineTools/usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -O2 -DNDEBUG -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/Users/dhruvmanilawala/contributing/neovim/build/config -I/Users/dhruvmanilawala/contributing/neovim/src -I/Users/dhruvmanilawala/contributing/neovim/.deps/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/usr/include -I/usr/local/opt/gettext/include -I/Users/dhruvmanilawala/contributing/neovim/build/src/nvim/auto -I/Users/dhruvmanilawala/contributing/neovim/build/include Compiled by dhruvmanilawala@mbp  Features: +acl +iconv +tui See ":help feature-compile"     system vimrc file: "$VIM/sysinit.vim"   fall-back for $VIM: "/Users/dhruvmanilawala/neovim/share/nvim"  Run :checkhealth for more info

### Operating system and version

macOS Big Sur 11.3

### Steps to reproduce

1. `nvim -nu minimal.lua`
2. Open the picker: `:Telescope current_buffer_fuzzy_find`

### Expected behavior

There should not be any error while highlighting the results buffer in `current_buffer_fuzzy_find`.

### Actual behavior

Printing the `hl_block` variable, we get the following two possibilities:
```
{ { 0, 1 }, "TelescopeResultsSpecialComment" }
{ { 10, 11 }, 854 }
```

As you can see, in the first case we get the highlight name but in the second case we are getting the highlight ID. This is due to this commit (https://github.com/neovim/neovim/commit/2460f0a7028550ea2d87492a4e8b95914fdba7b1) in the core which is now returning an ID (number) and `nvim_buf_add_highlight` does not like that.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      {
        "nvim-telescope/telescope.nvim",
        requires = {
          "nvim-lua/plenary.nvim",
          { "nvim-telescope/telescope-fzf-native.nvim", run = "make" },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
      { "nvim-treesitter/nvim-treesitter" },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require("telescope").setup()
  require("telescope").load_extension "fzf"
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
  require("nvim-treesitter.configs").setup {
    ensure_installed = { "lua" },
    highlight = { enable = true },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing Telescope and dependencies."
  vim.fn.system {
    "git",
    "clone",
    "--depth=1",
    "https://github.com/wbthomason/packer.nvim",
    install_path,
  }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by dhruvmanila** [comment]
On a side note: The `Neovim version` block does not get surrounded by triple backticks. I surrounded them myself, you can look at the edit history.

**Comment by HearseDev** [comment]
Note: If you turn off nvim-treesitter, the issue disappears.

**Comment by dhruvmanila** [comment]
Yes, because it won't use tree-sitter for highlighting then, but that's just a temporary workaround.
